cpusupport/cpusupport_arm_aes.c

```diff
@@ -31,6 +35,20 @@ CPUSUPPORT_FEATURE_DECL(arm, aes)
 	supported = (capabilities & HWCAP2_AES) ? 1 : 0;
 #endif
 #endif /* CPUSUPPORT_HWCAP_GETAUXVAL */
+
+#if defined(CPUSUPPORT_HWCAP_ELF_AUX_INFO)
+	unsigned long capabilities;
+
+#if defined(__aarch64__)
+	if (elf_aux_info(AT_HWCAP, &capabilities, sizeof(unsigned long)))
```

Comment:
Looking at this, I'm wondering if we should have a `readcaps` macro which invokes either `getauxval` or `elf_aux_info` depending on the macros?